Symfony/CS/Tokenizer/TokensAnalyzer.php

```diff
@@ -210,6 +210,71 @@ public function isArrayMultiLine($index)
         return false;
     }
 
+    public function getFunctionAttributes($index)
+    {
+        $tokens = $this->tokens;
+        $token = $tokens[$index];
+
+        if (!$token->isGivenKind(T_FUNCTION)) {
+            throw new \LogicException('No T_FUNCTION at given index');
+        }
+
+        $attributes = array(
+            'visibility' => null,
+            'static' => false,
+            'abstract' => false,
+            'final' => false,
+        );
+
+        for ($i = $index; $i >= 0; --$i) {
+            $token = $tokens->getPrevMeaningfulToken($i);
```

Comment:
updated in https://github.com/SpacePossum/PHP-CS-Fixer/commit/04cc5348bee143648c9439937e12879009a49779